theia-extensions/viewer-prototype/src/browser/trace-viewer/trace-viewer-toolbar-contribution.tsx

```diff
@@ -0,0 +1,106 @@
+import { TraceViewerWidget } from './trace-viewer';
+import { TspClientProvider } from '../tsp-client-provider-impl';
+import { TraceExplorerViewsWidget } from '../trace-explorer/trace-explorer-sub-widgets/trace-explorer-views-widget';
+import { TraceViewerToolbarFilterMenus, TRACE_VIEWER_TOOLBAR_COMMAND_FILTER } from './trace-viewer-toolbar-commands';
+
+@injectable()
+export class TraceViewerToolbarContribution extends AbstractViewContribution<TraceViewerWidget> implements TabBarToolbarContribution {
+    @inject(ApplicationShell) protected readonly shell: ApplicationShell;
+    @inject(ContextMenuRenderer) protected readonly contextMenuRenderer!: ContextMenuRenderer;
+    @inject(TspClientProvider) protected readonly tspClientProvider!: TspClientProvider;
+    @inject(MenuModelRegistry)
+    protected readonly menus: MenuModelRegistry;
+    @inject(CommandRegistry)
+    protected readonly commands: CommandRegistry;
+    private outputAnnotationsMap: Map<string, { categoryName: string, toggleInd: boolean }[]> = new Map<string, { categoryName: string, toggleInd: boolean }[]>();
+    private onAnnotationsFetchedSignal = (payload: { [key: string]: { categoryName: string, toggleInd: boolean }[] }) => this.doHandleAnnotationsFetchedSignal(payload);
+    protected readonly onAnnotationsChangedEmitter = new Emitter<void>();
+    protected readonly onAnnotationsChangedEvent = this.onAnnotationsChangedEmitter.event;
+
+    constructor() {
+        super({
+            widgetId: TraceExplorerViewsWidget.ID,
+            widgetName: TraceExplorerViewsWidget.LABEL,
+            defaultWidgetOptions: {
+                area: 'right',
+            },
+        });
+        signalManager().on(Signals.ANNOTATIONS_FETCHED, this.onAnnotationsFetchedSignal);
+    }
+
+    initializeLayout(): void {
+        this.openView({ activate: false });
+    }
+
+    private doHandleAnnotationsFetchedSignal(payload: { [key: string]: { categoryName: string, toggleInd: boolean }[] }) {
+        Object.entries(payload).forEach(([key, value]) => {
+            if (value.length === 0) {
+                this.outputAnnotationsMap.delete(key);
+            } else {
+                this.outputAnnotationsMap.set(key, value);
+            }
+        });
+        this.onAnnotationsChangedEmitter.fire();
+    }
+
+    registerToolbarItems(registry: TabBarToolbarRegistry): void {
+        registry.registerItem({
+            id: TRACE_VIEWER_TOOLBAR_COMMAND_FILTER.id,
+            isVisible: w => w instanceof TraceViewerWidget && this.outputAnnotationsMap.size>0,
+            // render() here is not a react component and hence need to disable the react display-name rule
+            // eslint-disable-next-line react/display-name
+            render: () => <div className="item enabled">
+                <div id="trace.viewer.toolbar.filter" className="fa fa-filter" title="Trace Viewer Toolbar filter" onClick={async (event: React.MouseEvent) => {
```

Comment:
The title 'Trace Viewer Toolbar filter' is not specific enough for the user. It should be something like 'Markers filter'

Reply:
Fixed